### ISSUE TITLE
prevent instance template from being updated when cos image is updated

### ISSUE THREAD

**Issue by bschaatsbergen** [bschaatsbergen/terraform-gce-atlantis#98]
When a new version of the cos image is available, it will now trigger a update on the instance template (as we always pull the latest version).

We should ignore any changes to this, and possibly allow a user to overwrite or set its own version of the COS image. As update the instance template will trigger Atlantis to be recreated. (this sucks, especially if this change is through Atlantis 😄 )

.... the plan I got

```
  # module.atlantis.google_compute_instance_template.default must be replaced
+/- resource "google_compute_instance_template" "default" {
      ~ id                   = "projects/xxxxxxxx/global/instanceTemplates/atlantis-20230210130232854100000001" -> (known after apply)
      ~ labels               = { # forces replacement
          ~ "container-vm" = "cos-stable-101-17162-127-5" -> "cos-stable-101-17162-127-8"
        }
      ~ metadata_fingerprint = "djl8in4QXsc=" -> (known after apply)
      - min_cpu_platform     = "" -> null
      ~ name                 = "atlantis-20230210130232854100000001" -> (known after apply)
      ~ self_link            = "https://www.googleapis.com/compute/v1/projects/xxxxxxxx/global/instanceTemplates/atlantis-20230210130232854100000001" -> (known after apply)
        tags                 = [
            "atlantis-wl45dn",
        ]
      ~ tags_fingerprint     = "" -> (known after apply)
        # (8 unchanged attributes hidden)

      + confidential_instance_config {
          + enable_confidential_compute = (known after apply)
        }

      ~ disk {
          ~ device_name       = "persistent-disk-0" -> (known after apply)
          ~ interface         = "SCSI" -> (known after apply)
          - labels            = {} -> null
          ~ mode              = "READ_WRITE" -> (known after apply)
          - resource_policies = [] -> null
          ~ source_image      = "projects/cos-cloud/global/images/cos-stable-101-17162-127-5" -> "https://www.googleapis.com/compute/v1/projects/cos-cloud/global/images/cos-stable-101-17162-127-8" # forces replacement
          ~ type              = "PERSISTENT" -> (known after apply)
            # (4 unchanged attributes hidden)
        }
      ~ disk {
          ~ boot              = false -> (known after apply)
          ~ interface         = "SCSI" -> (known after apply)
          - labels            = {} -> null
          - resource_policies = [] -> null
          + source_image      = (known after apply)
          ~ type              = "PERSISTENT" -> (known after apply)
            # (5 unchanged attributes hidden)
        }

      ~ network_interface {
          + ipv6_access_type   = (known after apply)
          ~ name               = "nic0" -> (known after apply)
          ~ network            = "https://www.googleapis.com/compute/v1/projects/xxxxxxxx/global/networks/network" -> (known after apply)
          - queue_count        = 0 -> null
          + stack_type         = (known after apply)
          ~ subnetwork         = "https://www.googleapis.com/compute/v1/projects/xxxxxxxx/regions/europe-west4/subnetworks/subnetwork" -> "projects/xxxxxxxx/regions/europe-west4/subnetworks/subnetwork"
            # (1 unchanged attribute hidden)
        }

      ~ scheduling {
          - min_node_cpus       = 0 -> null
            # (4 unchanged attributes hidden)
        }

        # (2 unchanged blocks hidden)
    }
```

**Comment by bschaatsbergen** [comment]
wontfix, this module has specifically been built to make changes like this possible. It's worth more to regularly update the COS image, then avoiding an update. Atlantis instances created using this module have the ability to be taken down and booted up again through the managed instance group, as a persistent disk is used to store the data on.

**Comment by bschaatsbergen** [comment]
Fixed in #112
